multiqc/modules/nonpareil/nonpareil.py

```diff
@@ -0,0 +1,295 @@
+
+        # Add plots
+        self.nonpareil_redundancy_plot()
+
+    def parse_nonpareil_json(self, f):
+        """Parse the JSON output from nonpareil"""
+        import json
+
+        try:
+            json_raw = json.load(f["f"])
+        except json.JSONDecodeError as e:
+            log.warning(f"Could not parse nonpareil JSON: '{f['fn']}': {e}, skipping file")
+            return None, {}
+
+        # x.obs Rarefied sequencing effort.
+        # x.adj Adjusted rarefied sequencing effort.
+        # y.red Rarefied redundancy (observed).
+        # y.cov Rarefied coverage (corrected).
+        # y.sd Standard deviation of rarefied coverage.
+        # y.p25 Percentile 25 (1st quartile) of rarefied coverage.
+        # y.p50 Percentile 50 (median) of rarefied coverage.
+        # y.p75 Percentile 75 (3rd quartile) of rarefied coverage
+        if self.disp_type:
+            log.debug(f"Plotting dispersion for {self.disp_type}")
+
+        for s_name, data in json_raw.items():
+            # Convert base pairs to megabase pairs
+            data["x.adj"] = [(1e-6 if x == 0 else x) / 1000 / 1000 for x in data["x.adj"]]
+            data["x.model"] = [x / 1000 / 1000 for x in data["x.model"]]
+            # Convert fraction to percentage
+            data["y.cov"] = [x * 100 for x in data["y.cov"]]
+            data["y.model"] = [x * 100 for x in data["y.model"]]
+            data["y.sd"] = [x * 100 for x in data["y.sd"]]
+            data["y.p75"] = [x * 100 for x in data["y.p75"]]
+            data["y.p25"] = [x * 100 for x in data["y.p25"]]
+            # Prepare plot data
+            data["observed"] = {x: y for x, y in zip(data["x.adj"], data["y.cov"])}
+            data["model"] = {x: y for x, y in zip(data["x.model"], data["y.model"])}
+            # Calculate dispersion
+            # from https://github.com/lmrodriguezr/nonpareil/blob/162f1697ab1a21128e1857dd87fa93011e30c1ba/utils/Nonpareil/R/Nonpareil.R#L306-L318
+            disp_add = None
+            if self.disp_type == "sd":
+                disp_add = np.array(data["y.sd"])
+            elif self.disp_type == "ci95":
+                disp_add = np.array(data["y.sd"]) * 1.9
+            elif self.disp_type == "ci90":
+                disp_add = np.array(data["y.sd"]) * 1.64
+            elif self.disp_type == "ci50":
+                disp_add = np.array(data["y.sd"]) * 0.67
+            elif self.disp_type == "iq":
+                data["disp_upper"] = [[x, y] for x, y in zip(data["x.adj"], data["y.p75"])]
+                data["disp_lower"] = [[x, y] for x, y in zip(data["x.adj"], data["y.p25"])]
+            if disp_add is not None:
+                data["disp_upper"] = [[x, y] for x, y in zip(data["x.adj"], np.array(data["y.cov"]) + disp_add)]
+                data["disp_lower"] = [[x, y] for x, y in zip(data["x.adj"], np.array(data["y.cov"]) - disp_add)]
+
+        return json_raw
+
+    def nonpareil_general_stats_table(self):
```

Comment:
Even though most columns are hidden, this feels like too many for the General Statistics table.

My suggestion would be to have a Nonpareil-specific table with all of these, most can then be shown by default. Then just pick one or two that would be particularly useful to have alongside outputs from other bioinformatics tools to keep in the General Stats table.